### PR TITLE
Add test filtering to performance pull requests

### PR DESCRIPTION
**Pull request by matt-kwong**

### CODE REVIEW

File: tools/jenkins/run_performance.sh [hunk 1]
```diff
@@ -37,4 +37,13 @@ BENCHMARKS_TO_RUN="bm_closure bm_cq bm_call_create bm_error bm_chttp2_hpack bm_c
 # Enter the gRPC repo root
 cd $(dirname $0)/../..
 
-tools/profiling/microbenchmarks/bm_diff.py -d origin/$ghprbTargetBranch -b $BENCHMARKS_TO_RUN
+AFFECTS_C_CPP=`python -c 'import sys; \
```
Comment (ctiller) [inline tools/jenkins/run_performance.sh]: I'd recommend isolating this into a script somewhere... 

It's likely to get tweaked by people over time, and copy and pasted differently in different places, resulting in a maintenance nightmare.

File: tools/run_tests/python_utils/filter_pull_request_tests.py [hunk 2]
```diff
@@ -127,6 +127,9 @@ def add_trigger(self, trigger):
   'setup\.py$': [_PYTHON_TEST_SUITE]
 }
 
+# Regex that combines all keys in _WHITELIST_DICT
+_ALL_TRIGGERS = "(" + ")|(".join(_WHITELIST_DICT.keys()) + ")"
```
Comment (ctiller) [inline tools/run_tests/python_utils/filter_pull_request_tests.py]: Why not `"()|(" + ...`

Comment (matt-kwong) [inline tools/run_tests/python_utils/filter_pull_request_tests.py]: This creates a string that is `(regex1)|(regex2)|(regex3)...`

File: tools/jenkins/run_performance.sh (hunk 1 above)
Comment (matt-kwong) [inline tools/jenkins/run_performance.sh]: I added `tools/jenkins/run_c_cpp_test.sh` and the idea is to have Jenkins run `tools/jenkins/run_c_cpp_test.sh tools/jenkins/run_performance.sh`. This should make it easy to extend the filtering to any future jobs specific to C/C++.

**Comment by matt-kwong** [comment]
Ping - a queue for performance tests is starting to build up after enabling more microbenchmarks.

**Pull request merged**